board-together/FE-Board-Together
# Spencer/query/create user

#### What's this PR do?
This merge will:
- Add a "create account" button on the login page which will...
  - Conditionally render a create user login form
  - Allow the user to create a username
  - Ensure username is correct and does not already exist with error handling
- Fix all Cypress tests to incorporate the new login page logic

Additionally...
- A user visiting the page can click the 'demo' button in order to proceed as a random user. This will allow the visitors to proceed without having to create a username.

#### Where should the reviewer start?
After ensuring that the repo has been cloned locally...
- `cd` into the Board Together directory
- `git fetch`
- `git checkout `spencer/query/create-user`
#### How should this be manually tested?
In CLI run `npm start` which will automatically open the page in the browser. From here, the application changes can manually be tested.
In CLI run `npm run cypress` which will pull up the Cypress interface. Select the Chrome option and proceed to run all tests to ensure they pass.

#### Any background context you want to provide?
There are currently unused endpoints for the API including user creation and user deletion. This branch will add the use of the user creation endpoint. A future branch will add the user deletion functionality.

#### Screenshots
<img width="664" alt="image" src="https://user-images.githubusercontent.com/74210902/225004736-d1a6f44a-f7d6-471e-bf9d-5c0a46bed098.png">
<img width="655" alt="image" src="https://user-images.githubusercontent.com/74210902/225004795-18819746-ba79-47d8-8ea3-721ecd4c2a2c.png">
<img width="672" alt="image" src="https://user-images.githubusercontent.com/74210902/225012928-b62555f7-6a62-4e99-8dc4-ecc443234ed9.png">
<img width="651" alt="image" src="https://user-images.githubusercontent.com/74210902/225013105-ac79fab8-5a88-4b5e-83d3-ae2affb8ba83.png">
